src/tribler/core/components/libtorrent/tests/test_download_manager.py

```diff
@@ -238,39 +261,59 @@ async def test_start_download_existing_handle(fake_dlmgr):
     """
     Testing the addition of a torrent to the libtorrent manager, if there is a pre-existing handle.
     """
-    infohash = b'a' * 20
-
-    mock_handle = MagicMock()
-    mock_handle.info_hash = lambda: hexlify(infohash)
-    mock_handle.is_valid = lambda: True
+    infohash = b"a" * 20
 
-    mock_ltsession = MagicMock()
-    mock_ltsession.get_torrents = lambda: [mock_handle]
+    mock_handle = MagicMock(
+        info_hash=MagicMock(return_value=hexlify(infohash)),
+        is_valid=MagicMock(return_value=True),
+    )
+    mock_ltsession = MagicMock(get_torrents=MagicMock(return_value=[mock_handle]))
 
-    fake_dlmgr.get_session = lambda *_: mock_ltsession
+    fake_dlmgr.get_session = MagicMock(return_value=mock_ltsession)
 
-    download = await fake_dlmgr.start_download(tdef=TorrentDefNoMetainfo(infohash, 'name'), checkpoint_disabled=True)
+    download = await fake_dlmgr.start_download(
+        tdef=TorrentDefNoMetainfo(infohash, "name"), checkpoint_disabled=True
+    )
     handle = await download.get_handle()
     assert handle == mock_handle
     fake_dlmgr.downloads.clear()
     await download.shutdown()
 
 
+def test_convert_rate():
+    assert DownloadManager.convert_rate(0) == -1  # 0 is a special value
+    assert DownloadManager.convert_rate(-1) == 1  # -1 is a special value
+    assert DownloadManager.convert_rate(1) == 1024
+    assert DownloadManager.convert_rate(2) == 2048
+    assert DownloadManager.convert_rate(-2) == -2048
+
+
+def test_reverse_convert_rate():
+    assert DownloadManager.reverse_convert_rate(-1) == 0  # -1 is a special value
+    assert DownloadManager.reverse_convert_rate(1) == -1  # 1 is a special value
+
+    assert DownloadManager.reverse_convert_rate(0) == 0
+    assert DownloadManager.reverse_convert_rate(2) == 0
+    assert DownloadManager.reverse_convert_rate(-2) == -1
```

Comment:
It's the same issue again. If you had applied the suggested solution from my original comment, it would have been fixed.

Please consider using numbers that actually represent the `reverse_convert_rate`, e.g.:

```python
    assert DownloadManager.reverse_convert_rate(1024) == 1
    assert DownloadManager.reverse_convert_rate(2048) == 2
    assert DownloadManager.reverse_convert_rate(-2048) == -2
```

Reply:
@drew2a This?
```py
    for test_num in [0, 2, -2]:
        assert DownloadManager.reverse_convert_rate(test_num) == test_num // 1024
```
If so, I misread your previous review. My bad.

Reply:
Found it:
```py
def test_convert_rate():
    assert DownloadManager.convert_rate(0) == -1  # 0 is a special value
    assert DownloadManager.convert_rate(-1) == 1  # -1 is a special value
    assert DownloadManager.convert_rate(1) == 1024
    assert DownloadManager.convert_rate(2) == 2048
    assert DownloadManager.convert_rate(-2) == -2048


def test_reverse_convert_rate():
    assert DownloadManager.reverse_convert_rate(-1) == 0  # -1 is a special value
    assert DownloadManager.reverse_convert_rate(1) == -1  # 1 is a special value
    assert DownloadManager.reverse_convert_rate(0) == 0
    assert DownloadManager.reverse_convert_rate(1024) == 1
    assert DownloadManager.reverse_convert_rate(-2048) == -2
```
Will fix them.

Reply:
Done.